app/src/main/java/net/vonforst/evmap/api/nobil/NobilApi.kt

```diff
@@ -0,0 +1,215 @@
+package net.vonforst.evmap.api.nobil
+
+import android.content.Context
+import com.car2go.maps.model.LatLng
+import com.car2go.maps.model.LatLngBounds
+import com.squareup.moshi.Moshi
+import net.vonforst.evmap.BuildConfig
+import net.vonforst.evmap.addDebugInterceptors
+import net.vonforst.evmap.api.ChargepointApi
+import net.vonforst.evmap.api.ChargepointList
+import net.vonforst.evmap.api.FiltersSQLQuery
+import net.vonforst.evmap.api.StringProvider
+import net.vonforst.evmap.model.ChargeLocation
+import net.vonforst.evmap.model.ChargepointListItem
+import net.vonforst.evmap.model.Filter
+import net.vonforst.evmap.model.FilterValue
+import net.vonforst.evmap.model.FilterValues
+import net.vonforst.evmap.model.ReferenceData
+import net.vonforst.evmap.model.getBooleanValue
+import net.vonforst.evmap.model.getSliderValue
+import net.vonforst.evmap.viewmodel.Resource
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import retrofit2.HttpException
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.POST
+import java.io.IOException
+import java.time.Duration
+
+private const val maxResults = 2000
+
+interface NobilApi {
+    @POST("search.php")
+    suspend fun getChargepoints(
+        @Body request: NobilRectangleSearchRequest
+    ): Response<NobilResponseData>
+
+    @POST("search.php")
+    suspend fun getChargepointsRadius(
+        @Body request: NobilRadiusSearchRequest
+    ): Response<NobilResponseData>
+
+    @POST("search.php")
+    suspend fun getChargepointDetail(
+        @Body request: NobilDetailSearchRequest
+    ): Response<NobilResponseData>
+
+    companion object {
+        private val cacheSize = 10L * 1024 * 1024 // 10MB
+
+        private val moshi = Moshi.Builder()
+            .add(LocalDateTimeAdapter())
+            .add(CoordinateAdapter())
+            .build()
+
+        fun create(
+            baseurl: String = "https://nobil.no/api/server/",
+            context: Context? = null
+        ): NobilApi {
+            val client = OkHttpClient.Builder().apply {
+                addInterceptor { chain ->
+                    val original = chain.request()
+                    val new = original.newBuilder()
+                        .build()
+                    chain.proceed(new)
+                }
```

Comment:
This interceptor is not necessary here - in other APIs it is used to automatically add the API key to every request, but here the API key is part of the request body so that's not really possible.